test/smokes/smoke.rb

```diff
@@ -109,11 +110,19 @@ def run
 
       def run_test(params, out)
         command_output, _ = Dir.mktmpdir do |dir|
-          repo_dir, base, head = prepare_git_repository(
-            workdir: Pathname(dir).realpath,
-            smoke_target: expectations.parent.join(params.name).realpath,
-            out: out,
-          )
+          if params.use_git_metadata
+            repo_dir, base, head = setup_existing_git_repository(
+                workdir: Pathname(dir).realpath,
+                smoke_target: expectations.parent.join(params.name).realpath,
+                out: out,
+                )
+          else
+            repo_dir, base, head = prepare_git_repository(
+                workdir: Pathname(dir).realpath,
+                smoke_target: expectations.parent.join(params.name).realpath,
+                out: out,
+                )
+          end
```

Comment:
[suggestion] How about the following refactoring?

```suggestion
          repo_args = {
            workdir: Pathname(dir).realpath,
            smoke_target: expectations.parent.join(params.name).realpath,
            out: out,
          }
          repo_dir, base, head =
            if params.use_git_metadata
              setup_existing_git_repository(**repo_args)
            else
              prepare_git_repository(**repo_args)
            end
```